lib/src/realm_class.dart

```diff
@@ -134,41 +134,28 @@ class Realm implements Finalizable {
   /// When the configuration is [LocalConfiguration] this completes right after the local [Realm] is opened or if the operation is canceled in advanced.
   /// Since opening a local Realm is a synchronous operation, there is no benefit of using Realm.open asynchronouse.
   static Future<Realm> open(Configuration config, {CancellationToken? cancellationToken, ProgressCallback? onProgressCallback}) async {
-    final cancellableCompleter = CancellableCompleter<Realm>(cancellationToken);
-    if (!cancellableCompleter.isCancelled) {
-      try {
-        final realm = Realm(config);
-        cancellableCompleter.future.catchError((Object error) {
-          realm.close();
-          return realm;
-        });
-
-        if (config is FlexibleSyncConfiguration) {
-          final session = realm.syncSession;
-          StreamSubscription<SyncProgress>? subscription;
-          try {
-            if (onProgressCallback != null) {
-              subscription = session
-                  .getProgressStream(
-                    ProgressDirection.download,
-                    ProgressMode.forCurrentlyOutstandingWork,
-                  )
-                  .listen(
-                    (syncProgress) => onProgressCallback.call(syncProgress),
-                    cancelOnError: true,
-                  );
-            }
-            await session.waitForDownload(cancellationToken);
-          } finally {
-            await subscription?.cancel();
-          }
-        }
-        cancellableCompleter.complete(realm);
-      } catch (error) {
-        cancellableCompleter.completeError(error);
+    final realm = Realm(config);
+    try {
+      if (config is FlexibleSyncConfiguration) {
+        final session = realm.syncSession;
+        await Future.wait<void>([
+          session.waitForDownload(cancellationToken),
+          if (onProgressCallback != null)
+            session
+                .getProgressStream(
+                  ProgressDirection.download,
+                  ProgressMode.forCurrentlyOutstandingWork,
+                )
+                .forEach(onProgressCallback)
+                .asCancellable(cancellationToken),
```

Comment:
I think that it is better to call subscribtion.cancel(). asCancellable would not call realmCore.sessionUnregisterConnectionStateNotifier.

Reply:
Yes it will. Why do you think that?